src/GitHub/Render.hs

```diff
@@ -0,0 +1,94 @@
+{- |
+Copyright: (c) 2020 Kowainik
+SPDX-License-Identifier: MPL-2.0
+Maintainer: Kowainik <xrom.xkov@gmail.com>
+
+GraphQL AST renderer.
+-}
+
+module GitHub.Render
+    ( renderTopQuery
+    , renderQuery
+    , renderQueryNode
+    , renderNodeName
+    , renderQueryParam
+    , renderParamName
+    , renderParamValue
+    , renderState
+    ) where
+
+import Data.List.NonEmpty (NonEmpty (..))
+import Data.Semigroup (stimes)
+import Data.Text (Text)
+
+import GitHub.GraphQL (NodeName (..), ParamName (..), ParamValue (..), Query (..), QueryNode (..),
+                       QueryParam (..), State (..))
+
+import qualified Data.Text as T
+
+
+renderTopQuery :: Query -> Text
+renderTopQuery q = "query {\n" <> renderQuery 1 q <> "\n}\n"
+
+renderQuery :: Int -> Query -> Text
+renderQuery i Query{..} = T.intercalate "\n"
+    $ map (renderQueryNode i) unQuery
+
+renderQueryNode :: Int -> QueryNode -> Text
+renderQueryNode i QueryNode{..} =
+    tab i <> renderNodeName queryNodeName
+    <> memptyIfFalse (null queryNodeArgs)
+       (between "(" ")" (T.intercalate ", " $ map renderQueryParam queryNodeArgs))
+    <> memptyIfFalse (null $ unQuery queryNode)
+       ( between " {\n" ("\n" <> tab i <> "}")
+           (T.intercalate "\n" $ map (renderQueryNode (i + 1)) $ unQuery queryNode)
+       )
+
+renderNodeName :: NodeName -> Text
+renderNodeName = \case
+    NodeRepository   -> "repository"
+    NodeIssues       -> "issues"
+    NodePullRequests -> "pullRequests"
+    NodeViewer       -> "viewer"
+    NodeTitle        -> "title"
+    NodeAuthor       -> "author"
+    NodeLogin        -> "login"
+    NodeResourcePath -> "resourcePath"
+    NodeUrl          -> "url"
+    NodeNodes        -> "nodes"
+    NodeEdges        -> "edges"
+
+renderQueryParam :: QueryParam -> Text
+renderQueryParam QueryParam{..} =
+    renderParamName queryParamName
+    <> ": "
+    <> renderParamValue queryParamValue
+
+renderParamName :: ParamName -> Text
+renderParamName = \case
+    ParamOwner  -> "owner"
+    ParamName   -> "name"
+    ParamLast   -> "last"
+    ParamStates -> "states"
+
+renderParamValue :: ParamValue -> Text
+renderParamValue = \case
+    ParamStringV str -> "\"" <> str <> "\""
```

Comment:
I wonder, do we need to do `show` here just in case to escape `"` inside the string properly?
```suggestion
    ParamStringV str -> T.pack $ show str
```